### PR TITLE
Use new Check For Updates service

### PR DESCRIPTION
**Pull request by psiinon**
Ready for review but marked as WIP as I think it would be better to merge just after then next weekly - that way we just get dev requests coming in initially as opposed to weekly ones.

Note that there no longer a fallback or insecure option, its the new service or nothing ;)

There are 2 commits in this PR - one for the CFU service changes and the other for SAST fixes - there were so many I couldnt leave them all :P

### CODE REVIEW

**Comment by psiinon** [comment]
Arg, missed some Constant changes...

**Comment by psiinon** [comment]
OK, so 3 commits now :P

File: zap/src/main/java/org/parosproxy/paros/Constant.java [hunk 1]
```diff
@@ -185,6 +185,15 @@
 
     private static final String VERSION_ELEMENT = "version";
 
+    public static enum OS {
+        Windows,
+        Linux,
+        MacOS,
+        Kali,
+        BackBox,
```
Comment (kingthorin) [inline zap/src/main/java/org/parosproxy/paros/Constant.java:193]: I've never heard of this?

File: zap/src/main/java/org/zaproxy/zap/extension/autoupdate/ExtensionAutoUpdate.java [hunk 2]
```diff
@@ -679,14 +652,11 @@ public void installNewExtensions() {
...
                                             logger.info(
-                                                    "Cant load add-on "
-                                                            + ao.getName()
-                                                            + " Not before="
-                                                            + ao.getNotBeforeVersion()
-                                                            + " Not from="
-                                                            + ao.getNotFromVersion()
-                                                            + " Version="
-                                                            + Constant.PROGRAM_VERSION);
+                                                    "Cant load add-on {} Not before={} Not from={} Version={}",
```
Comment (kingthorin) [inline zap/src/main/java/org/zaproxy/zap/extension/autoupdate/ExtensionAutoUpdate.java:655]:
```suggestion
                                                    "Can't load add-on: {} Not before={} Not from={} Version={}",
```

**Review by kingthorin — COMMENTED** [review]
Seems good so far. 

Finally using SonarLint?

**Comment by psiinon** [comment]
> Finally using SonarLint?

I am :D

File: zap/src/main/java/org/parosproxy/paros/Constant.java (hunk 1 above)
Comment (psiinon) [reply zap/src/main/java/org/parosproxy/paros/Constant.java:193]: https://www.backbox.org/ and they've updated to use the latest ZAP :)

Comment (kingthorin) [reply zap/src/main/java/org/parosproxy/paros/Constant.java:193]: Huh, okay cool.

File: zap/src/main/java/org/zaproxy/zap/extension/autoupdate/ExtensionAutoUpdate.java [hunk 3]
```diff
@@ -1174,48 +1175,17 @@ protected AddOnCollection getLatestVersionInfo(
...
-                                } else if (View.isInitialised()) {
-                                    shortUrl = ZAP_VERSIONS_REL_XML_DESKTOP_SHORT;
-                                    longUrl = ZAP_VERSIONS_REL_XML_FULL;
-                                } else {
-                                    shortUrl = ZAP_VERSIONS_REL_XML_DAEMON_SHORT;
-                                    longUrl = ZAP_VERSIONS_REL_XML_FULL;
-                                }
-                                boolean noInsecureUrlErrors = true;
-                                logger.debug("Getting latest version info from " + shortUrl);
+                                logger.debug("Getting latest version info from " + ZAP_CFU_SERVICE);
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/autoupdate/ExtensionAutoUpdate.java:1155]: Remove this? Already being logged in `getRemoteConfigurationUrl`.

File: zap/src/main/java/org/zaproxy/zap/extension/autoupdate/ExtensionAutoUpdate.java [hunk 4]
```diff
@@ -1174,48 +1175,17 @@ protected AddOnCollection getLatestVersionInfo(
...
-                                        if (callback != null) {
-                                            callback.insecureUrl(longUrl, e2);
-                                        }
-                                    } catch (Exception e2) {
-                                        logger.warn(
-                                                "Failed to check for updates using: " + longUrl,
-                                                e2);
-                                    }
+                                    logger.warn(
+                                            "Failed to check for updates using: " + ZAP_CFU_SERVICE,
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/autoupdate/ExtensionAutoUpdate.java:1164]: `… using: {}", ZAP_CFU_SERVICE`

**Comment by psiinon** [comment]
I _think_ I've fixed all of the comments...

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**